packages/frontend/src/components/message/MessageList.tsx

```diff
@@ -854,60 +854,81 @@ export const MessageListInner = React.memo(
       // over the lifetime of this component.
     })
 
-    if (!loaded) {
-      return (
-        <div id='message-list' ref={messageListRef} onScroll={onScroll2}>
-          <ol></ol>
-        </div>
-      )
-    }
-
     return (
-      <div id='message-list' ref={messageListRef} onScroll={onScroll2}>
+      <div
+        id='message-list'
+        ref={messageListRef}
+        onScroll={onScroll2}
+        // 'log' is appropriate for message lists.
+        // See https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA23.
+        role='log'
+        // `aria-live="polite"` is implied by `role="log"`.
+
+        // Otherwise screen readers will try to read
+        // the entire list of messages every time you switch the chat.
+        aria-busy={!loaded}
+        // Do not announce timestamp updates (see `updateTimestamps()`).
+        // We might want to reconsider this line
+        // when message editing gets implemented.
+        aria-relevant='additions'
+        // TODO how do we _not_ announce messages that get loaded
+        // due to scrolling / jumping to messages?
+        // Add `aria-hidden` for just one render cycle to such messages?
+        // Or apply `aria-busy` to this element when inserting such messages?
```

Comment:
OK, this really needs to get fixed. Because we also `fetchMoreTop` as soon as you open a chat, resulting in the screen readers announcing those newly inserted messages.

I think we might want to add something similar to `viewState`. A variable that would apply `aria-busy` for a single render. We would set it in `messagelist.ts` for effects such as `fetchMoreMessagesTop`.

Reply:
Another idea: apply `aria-live="off"` when the chat is not fully scrolled to the bottom.

Reply:
Also see https://github.com/deltachat/deltachat-desktop/issues/4743